### PR TITLE
Improve Mathlib

### PR DESCRIPTION
**Pull request by littlewhitecloud**

### CODE REVIEW

File: stdlib/math.jou [hunk 1]
```diff
@@ -68,14 +72,25 @@ declare asinh(x: double) -> double
...
+
+# Returns 2^x
 declare exp2(x: double) -> double
+declare exp2f(x: float) -> float
+
+# Returns 10^x
+declare exp10(x: double) -> double
+declare exp10f(x: float) -> float
+
 # Returns the natural logarithm of x.
```
Comment (Akuli) [inline stdlib/math.jou:87]: While we're here, we might as well clean up this comment. It only applies to `log()` and `logf()`. Other functions should have similar comments with the correct logarithm base, e.g. `# Returns the base 2 logarithm of x.` for `log2()` and `log2f()`.

File: stdlib/math.jou [hunk 2]
```diff
@@ -39,6 +40,9 @@ declare ceil(x: double) -> double
 declare floor(x: double) -> double
 # Returns the integral value that is nearest to x, with halfway cases rounded away from zero.
 declare round(x: double) -> double
```
Comment (Akuli) [inline stdlib/math.jou:42]: You could also add float versions of `ceil`, `floor`, `round` and `fabs`. (They just get `f` to end of name, e.g. the float version of `fabs` is `fabsf`.)

**Comment by Akuli** [comment]
This looks good, thanks for working on this!

I will "later" figure out why the CI is red. The changes don't look like they should break anything :)

File: stdlib/math.jou [hunk 3]
```diff
@@ -68,14 +72,25 @@ declare asinh(x: double) -> double
...
 declare exp2(x: double) -> double
+declare exp2f(x: float) -> float
+
+# Returns 10^x
+declare exp10(x: double) -> double
+declare exp10f(x: float) -> float
+
 # Returns the natural logarithm of x.
 declare log(x: double) -> double
+decalre logf(x: float) -> float
```
Comment (Akuli) [inline stdlib/math.jou:89]:
```suggestion
declare logf(x: float) -> float
```

**Review by Akuli — COMMENTED** [review]
Yeah this is why CI fails

File: stdlib/math.jou (hunk 2 above)
Comment (littlewhitecloud) [reply stdlib/math.jou:42]: Ok, I will add the float versions of these functions.

File: stdlib/math.jou (hunk 3 above)
Comment (littlewhitecloud) [reply stdlib/math.jou:89]: LOL, I spell `declare` keyword wrong. Thanks!

**Comment by Akuli** [comment]
Thanks for working on this! I pushed a couple commits and I will merge soon.

(If you are wondering why I deleted `fmod` and `fmodf`, it is because Jou's `%` operator already does the same thing. In C, the `%` operator is only for integers, and it cannot be used with floats or doubles.)

**Pull request merged**